### DISCUSSION TITLE
Does anyone else think the distinction between lora, lycoris, hypernetwork, embeddings seems pointless in extra networks?

### DISCUSSION THREAD

**Discussion by CapsAdmin** [vladmandic/sdnext#1547] [Ideas]
I've always thought this distinction in the UI seemed pointless. What benefit do I get from having these in separate tabs? They all modify the output in some way and I'm just left with the extra burden of switching tabs to find the right models. A model either goes in positive or negative prompt, so what type of model it is doesn't really concern me.

I appreciated the merge between LyCORIS and Lora as it made it easier to not have to switch tabs to find the right model. On the filesystem level it also made it easier to organize instead of having 2 separate folders like `lora/styles/somestyle.safetensors` and `lycoris/styles/someotherstyle.safetensors`

So I would personally take this a step further and merge hypernetworks and embeddings into a single tab and even the same folder on the filesystem.

Technically it should be possible to distinguish between these models but it would require reading the file. It should also be possible to just introduce a new directory where you can just place everything so you wouldn't break existing setups.

**Comment by tkiikt** [comment]
As the number of files grows, the next request may be to categorize them with tags and ratings, etc. Eventually, they will be classified in some way.

**Comment by arcum42** [comment]
What I really wish for is an option to either favorite them or sort by when they were last used. I tend to end up using the same ones over and over.

(You can somewhat do tags by putting them in subfolders...)

**Reply by rkiga** [reply]
You can kind of set favorites in the same way. Inside your e.g. lora folder, create a subfolder that comes first in the sorting order, like `@favs`. Then put your favorite lora in there.

**Comment by vladmandic** [comment]
while lora/locon/loha/lycoris/lokr/ia3 (yes, some may not even be that well known) are different types, they all belong to same family of networks - lora.
on the other hand, embeddings and hypernetworks are totally different.
while in general i'm all for simplificiation and i understand that end-user may not care about distinction, bundling everything together would be a step too far as they really do work totally differently.

what i have done in the latest update is to hide extra networks tab that do not have any extra networks. so for example, if you don't have or use hypernetworks, there is no reason why it should clutter the ui only to say "no networks found".

**Reply by CapsAdmin** [reply]
You don't necessarily have to hide the type information. One way would be to do it how civitai does it by allowing you to filter for certain types and have it shown as a tag.

![image](https://github.com/vladmandic/automatic/assets/204157/a8bbb983-40e2-47d3-afb1-97cfee026430)

**Reply by econundrum** [reply]
I was thinking that textual inversions work in a very different way that gives them different limitations and characteristics in use.

But I don't mind the tabs are merged so long as its easy to see what's a textual inversion.

Does anyone actually use hypernetworks? They don't seem as good as Loras or in someways even textual inversions.

Don't see many of them about, I actually don't even know how they work unlike textual inversion and Loras

**Reply by vladmandic** [reply]
re: hypernetworks - i stopped using them a while ago, don't see a point anymore.
all-in-all, having tag based display is a different way of organizing extra networks, valid suggestion, but given the current backlog, not that soon.

**Reply by econundrum** [reply]
I'd agree it shouldn't be a high priority to change this, there are as you say many more interesting or important things in the backlog.

**Reply by Tillerz** [reply]
The one thing with the tabs in "extra networks) that is annoying me is that the "models" are the first one. Why would anyone ever choose a model after going into extra networks. I would assume that LORAs ar the most commonly used ones, so having that tab be the default there would be a nice improvement. :)

**Comment by i-liek-katz** [comment]
I feel an "easy solution" would be an 'all' tab that is defaulted by settings if implemented. Could fill in the 'Extra networks tab order' box, or a tooltip with the network names so it's more apparent to end users on what it is and how to use it. Speaking of tooltip, there is a typo in the tooltip for 'Extra networks tab order' right now lsited instead of listed. Left-most would be the first "default" tab that appears first. Would also make it easy for users to remove tabs they just don't want a tab for.

**Reply by vladmandic** [reply]
they are so fundamentally different that i honestly don't want loras intertwined with embeddings. and "all" - i'm trying to imagine tab with checkpoints and loras all-in-one, how much miss-clicks causing model loads would that cause.

**Reply by i-liek-katz** [reply]
Yea, I personally like it separated. I don't want to see it become an "all" by default/only option, so proposed a middle ground that would also potentially work as a way for users to customize how they can use it. Currently can already reorder the tabs, but can't hide them. I'd personally like to hide checkpoints and hypernetworks, but best afaik is just reordering them.
